orchestrator-salt/src/main/resources/salt/salt/sssd/template/add_dns_record.j2

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo "$PW" | kinit {{ pillar['sssd-ipa']['principal'] }}
+
+set -x
+
+HOSTNAME=$(hostname)
+FQDN=$(hostname -f)
+IPADDR=$(hostname -i)
+REVERSE_IP=$(hostname -i | awk -F. '{print $4"."$3"." $2"."$1}')
+
+# dnsrecord-add must either add the record or modify it
+if ! ipa dnsrecord-find {{ pillar['sssd-ipa']['domain'] }}. "--name=${HOSTNAME}" "--a-rec=${IPADDR}"  --ttl {{ pillar['sssd-ipa']['dns_ttl'] }} --all; then
+  ipa dnsrecord-add {{ pillar['sssd-ipa']['domain'] }}. "${HOSTNAME}" "--a-rec=${IPADDR}" --ttl {{ pillar['sssd-ipa']['dns_ttl'] }}
+fi
+if ! ipa dnsrecord-find {{ pillar['sssd-ipa']['domain'] }}. "--name=${HOSTNAME}" "--a-rec=${IPADDR}"  --ttl {{ pillar['sssd-ipa']['dns_ttl'] }} --all; then
+  echo "Failed to set DNS A-record for ${HOSTNAME}"
+  false
+fi
+
+for zone in $(ipa dnszone-find --raw | grep idnsname | cut -d':' -f2 | xargs)
```

Comment:
I think you should grep for `.in-addr.arpa.` here

Reply:
also I have a little concern here. if we have lets see the IP `10.10.1.2`.
And we have 2 zones:
- `10.10.in-addr.arpa`
- `10.in-addr.arpa`
this for loop will add the reverse record for both zone. Is it really what we want to do?

Reply:
> I think you should grep for .in-addr.arpa. here
Done, but its technically not needed as `if echo "$REVERSE_IP" | grep -qE "$ZONE_NET$"; then` should have handled this.

also I have a little concern here. if we have

>  lets see the IP 10.10.1.2.
> And we have 2 zones:
> 
> 10.10.in-addr.arpa
> 10.in-addr.arpa
> this for loop will add the reverse record for both zone. Is it really what we want to do?
I don't think we should have cases where there are different matching zones, but maybe there is a way to do it with submitting with our current deployment of CDP. Either way, I made the reverse DNS register with the more specific zone. So it should only register once.

Reply:
> Done, but its technically not needed as...

Yes, but it will then loop for reverse zones only

We should not create but never know. Also customer might create it. I tested and FreeIPA will return the more exact match, but I think it's better to be prepared and add only for the right one.
Thanks for the modification!